blocks/basic/include/gnuradio-4.0/basic/ConverterBlocks.hpp

```diff
@@ -0,0 +1,267 @@
+#ifndef CONVERTERBLOCKS_HPP
+#define CONVERTERBLOCKS_HPP
+
+#include <tuple>
+
+#include <gnuradio-4.0/Block.hpp>
+#include <gnuradio-4.0/BlockRegistry.hpp>
+#include <gnuradio-4.0/DataSet.hpp>
+#include <gnuradio-4.0/meta/UncertainValue.hpp>
+
+namespace gr::blocks::type::converter {
+
+template<typename T, typename R>
+requires std::is_arithmetic_v<T> && std::is_arithmetic_v<R>
+struct Convert : public gr::Block<Convert<T, R>> {
+    using Description = Doc<"(@brief basic block to perform a input to output data type conversion (N.B. w/o scaling))">;
+    PortIn<T>  in;
+    PortOut<R> out;
+
+    template<gr::meta::t_or_simd<T> V>
+    [[nodiscard]] constexpr auto processOne(const V& input) const noexcept {
+        if constexpr (gr::meta::any_simd<V>) { // simd case
+            using RetType = std::experimental::rebind_simd_t<R, V>;
+            return vir::stdx::static_simd_cast<RetType>(input);
+        } else { // non-simd case
+            return static_cast<R>(input);
+        }
+    }
+};
+
+template<typename T, typename R>
+requires std::is_arithmetic_v<T> && std::is_arithmetic_v<R>
+struct ScalingConvert : public gr::Block<ScalingConvert<T, R>> {
+    using Description = Doc<R""(@brief basic block to perform a input to output data type conversion
+
+Performs scaling, i.e. 'R output = R(input * scale)'
+)"">;
+    PortIn<T>  in;
+    PortOut<R> out;
+    T          scale = static_cast<T>(1);
+
+    template<gr::meta::t_or_simd<T> V>
+    [[nodiscard]] constexpr auto processOne(const V& input) const noexcept {
+        if constexpr (gr::meta::any_simd<V>) { // simd case
+            using RetType = std::experimental::rebind_simd_t<R, V>;
+            return vir::stdx::static_simd_cast<RetType>(input * scale);
+        } else { // non-simd case
+            return static_cast<R>(input * scale);
+        }
+    }
+};
+
+template<typename T>
+requires std::is_arithmetic_v<T> || meta::complex_like<T>
+struct Abs : public gr::Block<Abs<T>> {
+    using R           = meta::fundamental_base_value_type_t<T>;
+    using Description = Doc<"@brief basic block to calculate the absolute (magnitude) value of a complex or arithmetic input stream">;
+    PortIn<T>  in;
+    PortOut<R> abs;
+
+    [[nodiscard]] constexpr R processOne(T input) const noexcept {
+        if constexpr (std::is_unsigned_v<T>) {
+            using TSigned = std::make_signed_t<T>;
+            return std::abs(static_cast<TSigned>(input));
```

Comment:
I think the return value has to be cast:
```
return static_cast<R>(std::abs(static_cast<TSigned>(input)));
```
Otherwise one has a lot of warnings. Example: (int - unsigned int), (double - float) etc.

Reply:
Done.